{"priority": "critical", "kind": "bug", "repository": {"links": {"self": {"href": "data/repositories/osrf/srcsim.json"}, "html": {"href": "#!/osrf/srcsim"}, "avatar": {"href": "data/bytebucket.org/ravatar/{c41021d9-0217-4628-999d-7f8d82f98dcb}ts=c_plus_plus"}}, "type": "repository", "name": "srcsim", "full_name": "osrf/srcsim", "uuid": "{c41021d9-0217-4628-999d-7f8d82f98dcb}"}, "links": {"attachments": {"href": "data/repositories/osrf/srcsim/issues/206/attachments_page=1.json"}, "self": {"href": "data/repositories/osrf/srcsim/issues/206.json"}, "watch": {"href": "https://api.bitbucket.org/2.0/repositories/osrf/srcsim/issues/206/watch"}, "comments": {"href": "data/repositories/osrf/srcsim/issues/206/comments_page=1.json"}, "html": {"href": "#!/osrf/srcsim/issues/206/gazebo-real-time-factor-falls-to-00-in"}, "vote": {"href": "https://api.bitbucket.org/2.0/repositories/osrf/srcsim/issues/206/vote"}}, "reporter": {"display_name": "Vinayak Jagtap", "uuid": "{4ca9a5d1-e34c-4476-9994-ccb04b8c4c73}", "links": {"self": {"href": "https://api.bitbucket.org/2.0/users/%7B4ca9a5d1-e34c-4476-9994-ccb04b8c4c73%7D"}, "html": {"href": "https://bitbucket.org/%7B4ca9a5d1-e34c-4476-9994-ccb04b8c4c73%7D/"}, "avatar": {"href": "data/secure.gravatar.com/avatar/d6b7a3e100816973cdc8ca51912fc868d=httpsavatar-management--avatars.us-west-2.prod.public.atl-paas.netinitialsVJ-4.png"}}, "nickname": "vinayak_jagtap", "type": "user", "account_id": "557058:e3f455ed-c5fc-439b-b24b-e6740163bb87"}, "title": "Gazebo real time factor falls to 0.0 in srcsim 0.6", "component": {"name": "robot", "links": {"self": {"href": "data/repositories/osrf/srcsim/components/395698.json"}}}, "votes": 0, "watches": 1, "content": {"raw": "Saw this error twice today. After getting the error, the real time factor is 0.0. In both cases, the robot was fixing either pitch or yaw in task1. Any specific log that I should be looking at to provide more details?\r\n\r\n```\r\n[ERROR] [1496438954.141746404, 279.640000000]: Exception in thread \"SynchronousMultiThreadedRobotController-thread-1\" java.lang.RuntimeException: java.lang.IndexOutOfBoundsException: Index: 29, Size: 0                                       \r\n[ERROR] [1496438954.141839560, 279.640000000]:  at us.ihmc.wholeBodyController.DRCControllerThread.run(DRCControllerThread.java:360)                                                                                                            \r\n[ERROR] [1496438954.141897458, 279.640000000]:  at us.ihmc.wholeBodyController.concurrent.SynchronousMultiThreadedRobotController$ReentrantLockedControlElementRunner.run(SynchronousMultiThreadedRobotController.java:89)                      \r\n[ERROR] [1496438954.141947921, 279.640000000]:  at java.lang.Thread.run(Thread.java:745)\r\n[ERROR] [1496438954.142116091, 279.640000000]: Caused by: java.lang.IndexOutOfBoundsException: Index: 29, Size: 0\r\n[ERROR] [1496438954.142157559, 279.640000000]:  at us.ihmc.robotics.lists.RecyclingArrayList.rangeCheck(RecyclingArrayList.java:340)                                                                                                            \r\n[ERROR] [1496438954.142204445, 279.640000000]:  at us.ihmc.robotics.lists.RecyclingArrayList.get(RecyclingArrayList.java:130)                                                                                                                   \r\n[ERROR] [1496438954.142251275, 279.640000000]:  at us.ihmc.robotics.math.trajectories.waypoints.FrameTrajectoryPointList.getTrajectoryPoint(FrameTrajectoryPointList.java:109)                                                                  \r\n[ERROR] [1496438954.142298453, 279.640000000]:  at us.ihmc.commonWalkingControlModules.controlModules.chest.TaskspaceChestControlState.queueExceedingTrajectoryPointsIfNeeded(TaskspaceChestControlState.java:260)                              \r\n[ERROR] [1496438954.142344671, 279.640000000]:  at us.ihmc.commonWalkingControlModules.controlModules.chest.TaskspaceChestControlState.initializeTrajectoryGenerator(TaskspaceChestControlState.java:230)                                       \r\n[ERROR] [1496438954.142392687, 279.640000000]:  at us.ihmc.commonWalkingControlModules.controlModules.chest.TaskspaceChestControlState.doAction(TaskspaceChestControlState.java:112)                                                            \r\n[ERROR] [1496438954.142446146, 279.640000000]:  at us.ihmc.robotics.stateMachines.conditionBasedStateMachine.GenericStateMachine.doAction(GenericStateMachine.java:152)                                                                         \r\n[ERROR] [1496438954.142495996, 279.640000000]:  at us.ihmc.commonWalkingControlModules.controlModules.chest.ChestOrientationManager.compute(ChestOrientationManager.java:126)                                                                   \r\n[ERROR] [1496438954.142544149, 279.640000000]:  at us.ihmc.commonWalkingControlModules.highLevelHumanoidControl.highLevelStates.WalkingHighLevelHumanoidController.updateManagers(WalkingHighLevelHumanoidController.java:595)\r\n[ERROR] [1496438954.142599356, 279.640000000]:  at us.ihmc.commonWalkingControlModules.highLevelHumanoidControl.highLevelStates.WalkingHighLevelHumanoidController.doAction(WalkingHighLevelHumanoidController.java:518)\r\n[ERROR] [1496438954.142662317, 279.640000000]:  at us.ihmc.robotics.stateMachines.conditionBasedStateMachine.GenericStateMachine.doAction(GenericStateMachine.java:152)\r\n[ERROR] [1496438954.142701400, 279.640000000]:  at us.ihmc.commonWalkingControlModules.highLevelHumanoidControl.HighLevelHumanoidControllerManager.doControl(HighLevelHumanoidControllerManager.java:178)\r\n[ERROR] [1496438954.142738952, 279.640000000]:  at us.ihmc.robotics.robotController.ModularRobotController.doControl(ModularRobotController.java:20)\r\n[ERROR] [1496438954.142775067, 279.640000000]:  at us.ihmc.wholeBodyController.DRCControllerThread.run(DRCControllerThread.java:352)\r\n[ERROR] [1496438954.142815993, 279.640000000]:  ... 2 more\r\n\r\n```", "markup": "markdown", "html": "<p>Saw this error twice today. After getting the error, the real time factor is 0.0. In both cases, the robot was fixing either pitch or yaw in task1. Any specific log that I should be looking at to provide more details?</p>\n<div class=\"codehilite\"><pre><span></span>[ERROR] [1496438954.141746404, 279.640000000]: Exception in thread &quot;SynchronousMultiThreadedRobotController-thread-1&quot; java.lang.RuntimeException: java.lang.IndexOutOfBoundsException: Index: 29, Size: 0                                       \n[ERROR] [1496438954.141839560, 279.640000000]:  at us.ihmc.wholeBodyController.DRCControllerThread.run(DRCControllerThread.java:360)                                                                                                            \n[ERROR] [1496438954.141897458, 279.640000000]:  at us.ihmc.wholeBodyController.concurrent.SynchronousMultiThreadedRobotController$ReentrantLockedControlElementRunner.run(SynchronousMultiThreadedRobotController.java:89)                      \n[ERROR] [1496438954.141947921, 279.640000000]:  at java.lang.Thread.run(Thread.java:745)\n[ERROR] [1496438954.142116091, 279.640000000]: Caused by: java.lang.IndexOutOfBoundsException: Index: 29, Size: 0\n[ERROR] [1496438954.142157559, 279.640000000]:  at us.ihmc.robotics.lists.RecyclingArrayList.rangeCheck(RecyclingArrayList.java:340)                                                                                                            \n[ERROR] [1496438954.142204445, 279.640000000]:  at us.ihmc.robotics.lists.RecyclingArrayList.get(RecyclingArrayList.java:130)                                                                                                                   \n[ERROR] [1496438954.142251275, 279.640000000]:  at us.ihmc.robotics.math.trajectories.waypoints.FrameTrajectoryPointList.getTrajectoryPoint(FrameTrajectoryPointList.java:109)                                                                  \n[ERROR] [1496438954.142298453, 279.640000000]:  at us.ihmc.commonWalkingControlModules.controlModules.chest.TaskspaceChestControlState.queueExceedingTrajectoryPointsIfNeeded(TaskspaceChestControlState.java:260)                              \n[ERROR] [1496438954.142344671, 279.640000000]:  at us.ihmc.commonWalkingControlModules.controlModules.chest.TaskspaceChestControlState.initializeTrajectoryGenerator(TaskspaceChestControlState.java:230)                                       \n[ERROR] [1496438954.142392687, 279.640000000]:  at us.ihmc.commonWalkingControlModules.controlModules.chest.TaskspaceChestControlState.doAction(TaskspaceChestControlState.java:112)                                                            \n[ERROR] [1496438954.142446146, 279.640000000]:  at us.ihmc.robotics.stateMachines.conditionBasedStateMachine.GenericStateMachine.doAction(GenericStateMachine.java:152)                                                                         \n[ERROR] [1496438954.142495996, 279.640000000]:  at us.ihmc.commonWalkingControlModules.controlModules.chest.ChestOrientationManager.compute(ChestOrientationManager.java:126)                                                                   \n[ERROR] [1496438954.142544149, 279.640000000]:  at us.ihmc.commonWalkingControlModules.highLevelHumanoidControl.highLevelStates.WalkingHighLevelHumanoidController.updateManagers(WalkingHighLevelHumanoidController.java:595)\n[ERROR] [1496438954.142599356, 279.640000000]:  at us.ihmc.commonWalkingControlModules.highLevelHumanoidControl.highLevelStates.WalkingHighLevelHumanoidController.doAction(WalkingHighLevelHumanoidController.java:518)\n[ERROR] [1496438954.142662317, 279.640000000]:  at us.ihmc.robotics.stateMachines.conditionBasedStateMachine.GenericStateMachine.doAction(GenericStateMachine.java:152)\n[ERROR] [1496438954.142701400, 279.640000000]:  at us.ihmc.commonWalkingControlModules.highLevelHumanoidControl.HighLevelHumanoidControllerManager.doControl(HighLevelHumanoidControllerManager.java:178)\n[ERROR] [1496438954.142738952, 279.640000000]:  at us.ihmc.robotics.robotController.ModularRobotController.doControl(ModularRobotController.java:20)\n[ERROR] [1496438954.142775067, 279.640000000]:  at us.ihmc.wholeBodyController.DRCControllerThread.run(DRCControllerThread.java:352)\n[ERROR] [1496438954.142815993, 279.640000000]:  ... 2 more\n</pre></div>", "type": "rendered"}, "assignee": null, "state": "new", "version": null, "edited_on": null, "created_on": "2017-06-02T21:37:49.840352+00:00", "milestone": null, "updated_on": "2017-06-10T01:20:45.680589+00:00", "type": "issue", "id": 206}